expression/expression.go

```diff
@@ -77,6 +84,54 @@ func (triggerExpression TriggerExpression) Get(name string) (interface{}, error)
 	}
 }
 
+// Validate returns error if triggers of type moira.ExpressionTrigger are badly formatted otherwise nil
+func (triggerExpression *TriggerExpression) Validate() error {
+	if triggerExpression.TriggerType != moira.ExpressionTrigger {
+		return fmt.Errorf("only advanced triggers should be validated")
+	}
+	if triggerExpression.Expression == nil {
+		return fmt.Errorf("trigger_type set to expression, but no expression provided")
+	}
+	expression := *triggerExpression.Expression
+	cacheKey := triggerExpression.validationCacheKey()
```

Comment:
Here I just want to clarify that I understood you correctly. You added this feature because the same expression can have different targets, right?

And further, if the expression has been checked:
- if there was an error, then you return it
- if there was no error, then why are you doing this: `validated = false`, if `validated` is not used anywhere else and why do you then delete this expression from the cache?

Also, if you chose the cache path, then maybe you should remove from the cache the expressions that change when the trigger is changed or deleted?

Reply:
Cache is no longer used but just to explain this scenario:

- if there is an error, it is returned
- if there is not but it has been checked `err.internalError` will also be nil and the function will also return with nil error.
- since we used a map containing interface{} to cache, we cannot guarantee the type, hence the check 
`err, ok := result.(verificationResult)`
and if the type cast was not successful, this means that the key is used to store value of another type in the cache, and this must be unexpected behavior, this is why I chose to delete it from the cache.